folium/map.py

```diff
@@ -454,25 +454,27 @@ class Popup(MacroElement):
 
     _template = Template(
         """
-        var {{this.get_name()}} = L.popup({{ this.options|tojavascript }});
-
-        {% for name, element in this.html._children.items() %}
-            {% if this.lazy %}
-                {{ this._parent.get_name() }}.once('click', function() {
-                    {{ this.get_name() }}.setContent($(`{{ element.render(**kwargs).replace('\\n',' ') }}`)[0]);
-                });
-            {% else %}
-                var {{ name }} = $(`{{ element.render(**kwargs).replace('\\n',' ') }}`)[0];
-                {{ this.get_name() }}.setContent({{ name }});
-            {% endif %}
-        {% endfor %}
-
-        {{ this._parent.get_name() }}.bindPopup({{ this.get_name() }})
-        {% if this.show %}.openPopup(){% endif %};
-
-        {% for name, element in this.script._children.items() %}
-            {{element.render()}}
-        {% endfor %}
+        {% macro script(this, kwargs) %}
```

Comment:
I'm not sure but this seems like a change that might break things. I know popups are a bit different from the rest. This is also a change I'd like to consider separately. And that would need proper testing.